core/client/src/main/java/alluxio/client/netty/NettyUnderFileSystemFileWriter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.netty;
+
+import alluxio.Constants;
+import alluxio.exception.ExceptionMessage;
+import alluxio.network.protocol.RPCErrorResponse;
+import alluxio.network.protocol.RPCFileWriteRequest;
+import alluxio.network.protocol.RPCFileWriteResponse;
+import alluxio.network.protocol.RPCMessage;
+import alluxio.network.protocol.RPCResponse;
+import alluxio.network.protocol.databuffer.DataByteArrayChannel;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Writer for an under file system file through a worker data server via Netty. This class does not
+ * hold any state specific to the file and can be used to write to different files, but not
+ * concurrently. This class does not keep lingering resources and does not need to be closed.
+ */
+@NotThreadSafe
+public final class NettyUnderFileSystemFileWriter {
```

Comment:
Probably not directly related with this PR, there's no unit tests for Netty related classes, right? what's the long-term plan to add Netty unit tests?